### ISSUE TITLE
User with custom resources can't print packing slips

### ISSUE THREAD

**Issue by joebusby** [magento/magento2#4426]
## Steps to reproduce

Magento upgrade to 2.0.5 (probably earlier versions too)
Create a new user role called "Shipping Clerk"
Check all boxes under "Custom Resources"
Create a new user, give them the "Shipping Clerk" role
Select an order and "Submit Shipment"
go back to the orders list, select the order and "Print Packing Slips" and 
## Expected result

PDF would be generated containing the packing slip
## Actual result

Access denied

Please try to sign out and sign in again.
If you continue to receive this message, please contact the store owner.

This is related to #3031, but I could not find anything specific to the PDF output on github.  At this point, shipping clerks can't print packing slips without having Resource Access of "All".  I tried on different systems and browsers.  Could it be a configuration problem?  Any Suggestions?

Thanks in advance.

Magento 2.0.5
Running 2600 products on 2.0.5 and debian 8 on an LXC container with 3gb, 6 cores and php 5.6

**Comment by Mulderua** [comment]
Hi @joebusby, thank you for reporting this issue.
We created internal ticket MAGETWO-54418. 

For immediate fix this problem you can add this function to \Magento\Sales\Controller\Adminhtml\Order\Pdfshipments

```
protected function _isAllowed()
{
    return $this->_authorization->isAllowed('Magento_Sales::shipment');
}
```

**Comment by joebusby** [comment]
@NadiyaS  @Mulderua   I modified the file: Interceptor.php  
  In the folder: var / generation / Magento / Sales / Controller / Adminhtml / Order / Pdfshipments / 

and it worked!   Thanks for the help.  I assume 1.2.0 will contain this fix?

**Comment by joebusby** [comment]
I have upgraded to 2.1.2 and shipping clerks still must run the report twice to get it to work.  So. .  it's workable now, but if it's easy to fix that, it would be great.  Thanks for your efforts!
At this time, a shipping clerk has to do this operation twice  after every time they are logged out (timeout) several times a day.  Much time is wasted.

**Comment by magento-engcom-team** [comment]
@joebusby, thank you for your report.
We've created internal ticket(s) MAGETWO-54418 to track progress on the issue.

**Comment by magento-engcom-team** [comment]
@joebusby, thank you for your report.
The issue is already fixed in 2.3.0, 2.2.2
